texar/core/attention_mechanism.py

```diff
@@ -941,3 +941,38 @@ class AttentionWrapperState(NamedTuple):
     attention_state: MaybeTuple[torch.Tensor]
     r"""A single or tuple of nested objects containing attention mechanism
     states for each attention mechanism."""
+
+    def clone(self, **kwargs):
```

Comment:
And I don't think the `clone` method is really necessary. It's only used once and this check is really redundant since we create the initial `AttentionWrapperState` based on the size of the provided `initial_state`.